packages/astro/test/env-secret.test.js

```diff
@@ -1,77 +1,92 @@
+		const response = await fixture.fetch('/');
+		assert.equal(response.status, 200);
+	});
 
-		it('works in dev', async () => {
-			writeFileSync(
-				new URL('./fixtures/astro-env-server-secret/.env', import.meta.url),
-				'KNOWN_SECRET=5',
-				'utf-8'
-			);
-			fixture = await loadFixture({
-				root: './fixtures/astro-env-server-secret/',
-			});
-			devServer = await fixture.startDevServer();
-			const response = await fixture.fetch('/');
-			assert.equal(response.status, 200);
+	it('builds without throwing', async () => {
+		fixture = await loadFixture({
+			root: './fixtures/astro-env-server-secret/',
+			output: 'server',
+			adapter: testAdapter({
+				env: {
+					KNOWN_SECRET: '123456',
+					UNKNOWN_SECRET: 'abc',
+				},
+			}),
 		});
+		await fixture.build();
+		assert.equal(true, true);
+	});
 
-		it('builds without throwing', async () => {
-			fixture = await loadFixture({
-				root: './fixtures/astro-env-server-secret/',
-				output: 'server',
-				adapter: testAdapter({
-					env: {
-						KNOWN_SECRET: '123456',
-						UNKNOWN_SECRET: 'abc',
-					},
-				}),
-			});
-			await fixture.build();
-			app = await fixture.loadTestAdapterApp();
-			assert.equal(true, true);
+	it('adapter can set how env is retrieved', async () => {
+		fixture = await loadFixture({
+			root: './fixtures/astro-env-server-secret/',
+			output: 'server',
+			adapter: testAdapter({
+				env: {
+					KNOWN_SECRET: '123456',
+					UNKNOWN_SECRET: 'abc',
+				},
+			}),
 		});
+		await fixture.build();
+		const app = await fixture.loadTestAdapterApp();
+		const request = new Request('http://example.com/');
+		const response = await app.render(request);
+		assert.equal(response.status, 200);
 
-		it('adapter can set how env is retrieved', async () => {
-			fixture = await loadFixture({
-				root: './fixtures/astro-env-server-secret/',
-				output: 'server',
-				adapter: testAdapter({
-					env: {
-						KNOWN_SECRET: '123456',
-						UNKNOWN_SECRET: 'abc',
-					},
-				}),
-			});
-			await fixture.build();
-			app = await fixture.loadTestAdapterApp();
-			const request = new Request('http://example.com/');
-			const response = await app.render(request);
-			assert.equal(response.status, 200);
+		const html = await response.text();
+		const $ = cheerio.load(html);
 
-			const html = await response.text();
-			const $ = cheerio.load(html);
+		const data = JSON.parse($('#data').text());
 
-			const data = JSON.parse($('#data').text());
+		assert.equal(data.KNOWN_SECRET, 123456);
+		assert.equal(data.UNKNOWN_SECRET, 'abc');
+	});
 
-			assert.equal(data.KNOWN_SECRET, 123456);
-			assert.equal(data.UNKNOWN_SECRET, 'abc');
+	it('fails if validateSecretsOnStart is enabled and secret is not set', async () => {
+		fixture = await loadFixture({
+			root: './fixtures/astro-env-server-secret/',
+			experimental: {
+				env: {
+					validateSecretsOnStart: true,
+				},
+			},
 		});
+
+		let error = null;
+		try {
+			devServer = await fixture.startDevServer();
```

Comment:
```suggestion
			devServer = await fixture.startDevServer();
			assert.fail();
```

Just making sure that we fail the test if no error was thrown

Reply:
If no error is thrown, the asserts below will catch that since error will not be an instance of `Error` right?

Reply:
Ideally, you would want to run the assertions inside the `catch` block, and `.fail` inside the `try` block. There's so need to run assertions if you know something is failing to begin with.

But if you want to run the assertion **regardless**, I suggest to use the `finally` block